spec/paperclip/io_adapters/data_uri_adapter_spec.rb

```diff
@@ -73,6 +73,71 @@
 
   end
 
+  context "an svg image" do
+    context "base64 encoded" do
+      before do
+        @contents = "data:image/svg+xml;base64,#{svg_base64_content}"
+        @subject = Paperclip.io_adapters.for(@contents)
+      end
+      it "returns a nondescript file name" do
+        assert_equal "data", @subject.original_filename
+      end
+
+      it "returns a content type" do
+        assert_equal "image/svg+xml", @subject.content_type
+      end
+      it "generates correct fingerprint after read" do
+        fingerprint = Digest::MD5.hexdigest(@subject.read)
+        assert_equal fingerprint, @subject.fingerprint
+      end
+
+      it "generates same fingerprint" do
+        assert_equal @subject.fingerprint, @subject.fingerprint
+      end
+
+      it 'accepts a content_type' do
+        @subject.content_type = 'image/svg+xml'
+        assert_equal 'image/svg+xml', @subject.content_type
+      end
+
+      it 'accepts an original_filename' do
+        @subject.original_filename = 'image.svg'
+        assert_equal 'image.svg', @subject.original_filename
+      end
+    end
+    context "utf8 encoded" do
+      before do
+        @contents = "data:image/svg+xml;charset=utf-8,#{URI.encode(svg_file_contents)}"
+        @subject = Paperclip.io_adapters.for(@contents)
+      end
+      it "returns a nondescript file name" do
+        assert_equal "data", @subject.original_filename
+      end
+
+      it "returns a content type" do
+        assert_equal "image/svg+xml", @subject.content_type
+      end
+      it "generates correct fingerprint after read" do
+        fingerprint = Digest::MD5.hexdigest(@subject.read)
+        assert_equal fingerprint, @subject.fingerprint
+      end
+
+      it "generates same fingerprint" do
+        assert_equal @subject.fingerprint, @subject.fingerprint
+      end
+
+      it 'accepts a content_type' do
+        @subject.content_type = 'image/svg+xml'
+        assert_equal 'image/svg+xml', @subject.content_type
+      end
+
+      it 'accepts an original_filename' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.